### PR TITLE
Add draft support for space data sources

### PR DESCRIPTION
**Pull request by cwang**
## Description

Support for data sources in spaces.

Fixes/Implements #(issue/feature)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor and code improvement (non-breaking change which improves code quality and/or performance)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation
- [ ] Tests
- [ ] Other chores such as maintenance

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration by modifying the list below.

- [ ] Test A
- [ ] Test B
- [ ] Test C

**Test Configuration**:

Please describe the test setup. List them below as bullet points.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings
- [ ] The commit message follows the convention of this project

### CODE REVIEW

File: source/docq/data_source/main.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        """Get the name of the data source."""
+        return self.name
+
+    @abstractmethod
+    def load(self, space: SpaceKey, configs: dict) -> List[Document]:
+        """Load the documents from the data source."""
+        pass
+
+
+SPACE_DATA_SOURCES = {
```
[the remark this answers is not in the document]
Comment (cwang) [reply source/docq/data_source/main.py:30]: This is a registry for all available data sources and their implementations. Will come into play shortly when the PR is ready.

File: source/docq/config.py [hunk 2]
```diff
@@ -7,12 +7,6 @@
 ENV_VAR_OPENAI_API_KEY = "OPENAI_API_KEY"
 
 
-LLM_MODELS = {
-    "OPENAI_CHAT": (["gpt-3.5-turbo", "gpt-4"], range(0, 2)),
-    "OPENAI": (["text-davinci-003", "text-davinci-002", "code-davinci-002"], range(0, 2)),
-}
```
Comment (janaka) [inline source/docq/config.py:13]: Why are we dropping this?

Comment (cwang) [reply source/docq/config.py:13]: Moved

File: source/docq/data_source/azure_blob_storage.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+from typing import List
+
+from llama_index import Document
+
+from ..domain import SpaceKey
+from .main import SpaceDataSource
+
+
+class AzureBlobStorage(SpaceDataSource):
```
[the remark this answers is not in the document]
Comment (cwang) [reply source/docq/data_source/azure_blob_storage.py:11]: Actually It is what MS called it :) https://azure.microsoft.com/en-gb/products/storage/blobs 
But I think it makes sense to use a shorter name given the pre-set, clear context.

[the remark this answers is not in the document]
Comment (janaka) [reply source/docq/data_source/azure_blob_storage.py:11]: So Blob Storage `account` has one or more `containers`. Container seems to be the equivalent to an AWS S3 bucket. 
Hence why I feel calling it a container is better so the scope is clear from name. 

A container contains objects and supports path type prefixing. 

image


https://learn.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction

[the remark this answers is not in the document]
Comment (janaka) [reply source/docq/data_source/azure_blob_storage.py:11]: So `AzureBlobStorageContainer` really 😀

[the remark this answers is not in the document]
Comment (janaka) [reply source/docq/data_source/azure_blob_storage.py:11]: We don't need debate to death. Not incorrect to call it BlobStorage.

Doesn't matter for the container we manage but for the option of an admin creating a space attached to a container they manage.

how it appears in the UI

image

**Review by janaka — APPROVED** [review]
Left some more info re type name for Azure blob.

**Comment by cwang** [comment]
Bring in some UI changes for space data source support:
Screenshot 2023-07-03 at 22 13 05
Screenshot 2023-07-03 at 22 12 50

Also splitting the page under "Shared Spaces" into two, as the newly introduced streamlit chat UI components don't play well under `st.tab` - having two pages is the easiest fix plus following the same pattern as those under `Personal Space`

**Review by janaka — APPROVED** [review]
Lgtm

**Pull request merged**